File: HomeStorage.Logic/Models/ProductModels/ProductCreateModel.cs
using Microsoft.AspNetCore.Http;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace HomeStorage.Logic.Models.ProductModels
{
    public class ProductCreateModel
    {
        [Required]
        public Guid LocationId { get; set; }
        [Required, MinLength(3)]
        public required string Name { get; set; }
        public string? Description { get; set; }
        public List<LookupModel<Guid?>> Categories { get; set; } = [];
        public DateTime? ExpirationDate { get; set; }
        public double? Amount { get; set; }
        [JsonIgnore]
        public IFormFile? Image { get; set; }
    }
}

File: frontend/src/models/product/productCreateModel.ts
import { Lookup } from "@/models/lookup";

export interface ProductCreateModel {
    locationId: string;
    name: string;
    description?: string;
    imageId?: string;
    categories: Lookup<string | undefined>[];
    expirationDate?: string;
    amount?: number;
    image?: File;
}
